### ISSUE TITLE
0.5.5: false positive for `const X = Namespace.Member` re-exported via `export { X }` (breaks shadcn/ui, Radix wrappers)

### ISSUE THREAD

**Issue by talKitron** [ArnaudBarre/eslint-plugin-react-refresh#116]
Since 0.5.5, a `const` initialized from a member expression (e.g. `const Tooltip = TooltipPrimitive.Root`) is classified as a non-component when exported via `export { X }`. If the same file also exports a genuine component, the mixed-export error fires on the member-expression ones.

This is the standard shadcn/ui component shape, so it affects every such file in a project (45 errors across ~10 files in mine).

### Reproduction

```tsx
import * as React from "react";
import * as P from "@radix-ui/react-tooltip";

const A = P.Root;
const B = React.forwardRef<HTMLDivElement>((_p, ref) => <div ref={ref} />);

export { A, B };
```

```js
// eslint.config.js
import tsParser from "@typescript-eslint/parser";
import reactRefresh from "eslint-plugin-react-refresh";

export default [{
  files: ["**/*.tsx"],
  languageOptions: { parser: tsParser, parserOptions: { ecmaFeatures: { jsx: true } } },
  plugins: { "react-refresh": reactRefresh },
  rules: { "react-refresh/only-export-components": ["error", { allowConstantExport: true }] },
}];
```

**Expected:** no error — both `A` and `B` are components.

**Actual on 0.5.5:**
```
5:10  error  Fast refresh only works when a file only exports components.
             Use a new file to share constants or functions between components
             react-refresh/only-export-components
```
The error points at `A`; `B` is not flagged.

### Versions

| Version | Result |
|---|---|
| 0.5.3 | clean |
| 0.5.4 | clean |
| 0.5.5 | error |

### Narrowing

The error requires **both** kinds of export in one file:

- only `const A = P.Root` exported → clean
- two member-expression consts, no component → clean
- only `forwardRef` component → clean
- **one of each → error on the member-expression one**

So the member-expression const is being treated as a non-component, which then makes the file a "mixed" export.

### Likely cause

#114 (`Fix SCREAMING_SNAKE_CASE constant exported via export { Name }`, follow-up to #110) changed specifier handling to resolve a local declarator's initializer and run it through `handleExportIdentifier(identifier, init)` rather than the name-only `reactComponentNameRE` fallback. That fixed the SCREAMING_SNAKE_CASE case, but a `MemberExpression` initializer now appears to fall through to "not a component" instead of being treated as unknown.

Both `Tooltip` (PascalCase) and its initializer are consistent with a component; the previous name-based check accepted it. Re-exporting a component from a namespace import is common well beyond shadcn — any thin wrapper around Radix, Headless UI, etc.

### Note

Inline `export const A = P.Root;` is **not** affected — only the `export { A }` specifier form.

**Comment by ArnaudBarre** [comment]
Thanks for the report, I will see how to make `export const A` works like `const A; export { A }`

**Comment by Etsija** [comment]
Nice, I saw the same error just a minute ago, came here, and noticed it's already reported! We got exactly the same problem from shadcn-ui's [`Chart`](https://ui.shadcn.com/docs/components/base/chart) component now.

**Comment by ArnaudBarre** [comment]
Fixed in [v0.5.6](https://github.com/ArnaudBarre/eslint-plugin-react-refresh/releases/tag/v0.5.6)!
